### PR TITLE
InstalledVersions: add more information about packages

### PR DESCRIPTION
**Pull request by VaclavPavek**
Added more information for listing / checks packages in APP
- is_dev
- is_abandoned

### CODE REVIEW

File: src/Composer/Repository/FilesystemRepository.php [hunk 1]
```diff
@@ -147,14 +147,16 @@ public function write($devMode, InstallationManager $installationManager)
                 }
 
                 $versions['versions'][$package->getName()] = array(
+                    'name' => $package->getName(),
                     'pretty_version' => $package->getPrettyVersion(),
                     'version' => $package->getVersion(),
                     'aliases' => array(),
                     'reference' => $reference,
+                    'is_dev' => $package->isDev(),
```
Comment (stof) [inline src/Composer/Repository/FilesystemRepository.php:155]: I don't think this makes sense. This is something derived from the version (note that this is **not** telling you whether it is a dev requirement)

File: src/Composer/Repository/FilesystemRepository.php [hunk 2]
```diff
@@ -147,14 +147,16 @@ public function write($devMode, InstallationManager $installationManager)
                 }
 
                 $versions['versions'][$package->getName()] = array(
+                    'name' => $package->getName(),
```
Comment (stof) [inline src/Composer/Repository/FilesystemRepository.php:150]: why duplicating something which is already the key in the array of versions ?

Comment (VaclavPavek) [reply src/Composer/Repository/FilesystemRepository.php:151]: Duplicated for compatibility schema with  [InstalledVersions::getRootPackage](https://github.com/composer/composer/blob/master/src/Composer/InstalledVersions.php#L148) which return package name as array key.

Comment (stof) [reply src/Composer/Repository/FilesystemRepository.php:151]: but the difference is that when asking for info about another package, you already know the name, as you passed it as argument

File: src/Composer/Repository/FilesystemRepository.php (hunk 1 above)
Comment (VaclavPavek) [reply src/Composer/Repository/FilesystemRepository.php:156]: Yeah, you're right.

File: src/Composer/Repository/FilesystemRepository.php (hunk 2 above)
Comment (VaclavPavek) [reply src/Composer/Repository/FilesystemRepository.php:151]: ok, rollback

**Comment by VaclavPavek** [comment]
Closing this issue, I will continue to parse `composer.lock` because I need more information about the package:

- important: is dev requirement, is `abandoned`
- less important: `license`, url of `homepage` or `support`

**Pull request closed without merging**